support-workers/src/main/scala/com/gu/support/workers/lambdas/CreateZuoraSubscription.scala

```diff
@@ -99,12 +99,13 @@ class ZuoraSubscriptionCreator(
     } yield HandlerResult(getEmailState(state, account, sub, paymentOrRedemptionData, maybeGiftRecipWithCode), info)
   }
 
-  private def maybeAddCode(billingPeriod: BillingPeriod, giftRecipient: GiftRecipient) =
+  private def maybeAddCode(billingPeriod: BillingPeriod, giftRecipient: GiftRecipient, today: () => LocalDate) =
     giftRecipient match {
       case digitalSubGiftRecipient: DigitalSubGiftRecipient =>
         val giftCode = giftCodeGeneratorService.generateCode(billingPeriod)
           .withLogging("Generated code for Digital Subscription gift")
-        DigitalSubGiftPurchase(digitalSubGiftRecipient, giftCode)
+        val lastRedemptionDate = today().plusMonths(GiftCodeValidator.expirationTimeInMonths).minusDays(1)
```

Comment:
I took off one day just because a zuora 12 month sub runs from e.g. 2020-10-14 00:00 to 2021-10-14 00:00 but if we add 12 months to today, it looks like we include today.  Also it's a nice fudge factor.